server/src/controllers/course.ts

```diff
@@ -20,12 +24,54 @@ export interface CourseData {
   evaluationInformation: LocalizedString
 }
 
+export interface InstanceData extends CourseData {
+  startingPeriod: string,
+  endingPeriod: string
+  startDate: Date,
+  endDate: Date,
+  courseType: string,
+  gradingType: string,
+  responsibleTeachers: Array<string>,
+}
+
 export enum Language {
   English = 'EN',
   Finnish = 'FI',
   Swedish = 'SV'
 }
 
+function parseSisuInstance (instance: SisuInstance): InstanceData {
+  return {
+    // instance id here is Sisu instance id (string)
```

Comment:
This should also be specified as a comment in `CourseData` that the ID can either represent our internal ID (number) or Sisu ID (string).

Reply:
I noticed that the Sisu API does not have/return unique id for the course itself, just for the course instance. Therefore I edited the `CourseData` interface `id` so it can be undefined when representing parsed sisu course data. What do you think? https://github.com/aalto-grades/base-repository/pull/126/commits/af44fb7adac4d0c89819dc85039d07bb5c87b9d7

Reply:
That's fine too as long as it's explained why the ID can be undefined, which is explained.